File: android/app/src/main/java/sol/vortexx/android/bootstrap/api/NodeDirectory.kt
package sol.vortexx.android.bootstrap.api

/**
 * Locates reachable Vortex nodes at first launch.
 *
 * Responsibility: turn "the user just opened the app, we have no config"
 * into a working base URL or a clear "can't reach anyone" signal.
 * Nothing else — parsing, storage and UI all live elsewhere.
 */
interface NodeDirectory {
    suspend fun probePrimary(): ProbeResult
    suspend fun probe(url: String): ProbeResult
}

sealed interface ProbeResult {
    data class Ok(val baseUrl: String, val version: String?) : ProbeResult
    data class Unreachable(val tried: String, val reason: String) : ProbeResult
}

File: ios/Modules/Sources/Bootstrap/api/NodeDirectory.swift
import Foundation

/// Locates reachable Vortex nodes at first launch.
///
/// Single responsibility: turn "the user just opened the app, we have
/// no config" into a working base URL or a clear "can't reach anyone"
/// signal. Parsing, storage, UI all live elsewhere.
public protocol NodeDirectory: Sendable {
    func probePrimary() async -> ProbeResult
    func probe(url: String) async -> ProbeResult
}

public enum ProbeResult: Sendable, Equatable {
    case ok(baseUrl: String, version: String?)
    case unreachable(tried: String, reason: String)
}
